c2corg_api/views/document.py

```diff
@@ -186,6 +191,11 @@ def _create_document(
         document = schema.objectify(document_in)
         document.document_id = None
 
+        if document.geometry:
```

Comment:
there are many cases of `None` geometry in our tests which were not passing so it is better to do have if statement somewhere - I moved it into the `validate_geometry_type` function so it isn't duplicated at least.